### PR TITLE
Verify streaming output instantly

### PR DESCRIPTION
**Pull request by mahmoudhanafy**
I used queues instead of Lists and compare with expected output during execution.

### CODE REVIEW

File: src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala [hunk 1]
```diff
@@ -38,8 +38,8 @@ import scala.reflect.ClassTag
  */
 // tag::collectResults[]
 class TestOutputStream[T: ClassTag](parent: DStream[T],
-  val output: ArrayBuffer[Seq[T]] = ArrayBuffer[Seq[T]]()) extends Serializable {
-  
+  val output: mutable.SynchronizedQueue[Seq[T]] = new mutable.SynchronizedQueue[Seq[T]]) extends Serializable {
```
Comment (holdenk) [inline src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala]: SynchronizedQueue is deprecated, we should avoid using it.

File: src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala [hunk 2]
```diff
@@ -182,11 +180,15 @@ private[holdenkarau] trait StreamingSuiteCommon extends Logging with SparkContex
...
-      outputStream: TestOutputStream[V],
-      ssc: TestStreamingContext,
-      numBatches: Int,
-      numExpectedOutput: Int
-    ): Seq[Seq[V]] = {
+     outputStream: TestOutputStream[V],
+     ssc: TestStreamingContext,
+     numBatches: Int,
+     expectedOutput: Seq[Seq[V]],
+     verifyOutputTillNow: (mutable.Queue[Seq[V]], mutable.Queue[Seq[V]]) => Unit
```
Comment (holdenk) [inline src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala]: This is a strange way to handle the comparison.

File: src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala [hunk 3]
```diff
@@ -182,11 +180,15 @@ private[holdenkarau] trait StreamingSuiteCommon extends Logging with SparkContex
...
   private[holdenkarau] def runStreams[V: ClassTag](
-      outputStream: TestOutputStream[V],
-      ssc: TestStreamingContext,
-      numBatches: Int,
-      numExpectedOutput: Int
-    ): Seq[Seq[V]] = {
+     outputStream: TestOutputStream[V],
+     ssc: TestStreamingContext,
+     numBatches: Int,
+     expectedOutput: Seq[Seq[V]],
```
Comment (holdenk) [inline src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala]: Why do we take this as a Seq if we are going to convert it to a mutable queue right away?

File: src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala (hunk 2 above)
Comment (mahmoudhanafy) [inline src/main/1.3/scala/com/holdenkarau/spark/testing/StreamingSuiteCommon.scala]: We are really collecting the entire thing, but we compare instantly. This means that may be before collecting the second DStream, the first Dstream could be matched then with the expected output.

**Pull request closed without merging**